sdks/python/apache_beam/io/gcp/experimental/spannerio.py

```diff
@@ -320,6 +330,42 @@ def __init__(self, spanner_configuration):
     self._spanner_configuration = spanner_configuration
     self._snapshot = None
     self._session = None
+    self.base_labels = {
+        monitoring_infos.SERVICE_LABEL: 'Spanner',
+        monitoring_infos.METHOD_LABEL: 'Read',
+        monitoring_infos.SPANNER_PROJECT_ID: self._spanner_configuration.
+        project,
+        monitoring_infos.SPANNER_DATABASE_ID: self._spanner_configuration.
+        database,
+    }
+
+  def _table_metric(self, table_id, status):
+    database_id = self._spanner_configuration.database
+    project_id = self._spanner_configuration.project
+    resource = resource_identifiers.SpannerTable(
+        project_id, database_id, table_id)
+    labels = {
+        **self.base_labels,
+        monitoring_infos.RESOURCE_LABEL: resource,
+        monitoring_infos.SPANNER_TABLE_ID: table_id
+    }
+    service_call_metric = ServiceCallMetric(
```

Comment:
ServiceCallMetric by design must be instantiated before the IO(spanner) API call is made to the IO source/sink.
This is because we intened to use it to also time the IO(spanner) API calls.

Then the .call() should be made once the IO(spanner) API call returns